homeassistant/components/toon/climate.py

```diff
@@ -1,90 +1,153 @@
-        self._temperature = None
-        self._setpoint = None
-        self._operation_list = [
-            STATE_AUTO,
-            STATE_HEAT,
-            STATE_ECO,
-            STATE_COOL,
-        ]
+
+        self._current_temperature = None
+        self._target_temperature = None
+        self._real_target_temperature = None
+        self._next_target_temperature = None
+        self._modulation_level = None
+
+        self._heating_type = None
+        self._program_state = None
+        self._program_next = None
+        self._holiday_state = None
+
+        super().__init__(toon, "Toon Thermostat", 'mdi:thermostat')
 
     @property
-    def supported_features(self):
-        """Return the list of supported features."""
-        return SUPPORT_FLAGS
+    def unique_id(self) -> str:
+        """Return the unique ID for this thermostat."""
+        return self.toon.agreement.id
 
     @property
-    def name(self):
-        """Return the name of this thermostat."""
-        return self._name
+    def device_info(self) -> Dict[str, Any]:
+        """Return device information about this thermostat."""
+        toon = self.toon.agreement
+        return {
+            'identifiers': {
+                (DOMAIN, toon.id)
+            },
+            'name': self._name,
+            'manufacturer': 'Eneco',
+            'model': toon.display_hardware_version.rpartition('/')[0],
+            'sw_version': toon.display_software_version.rpartition('/')[-1],
+        }
+
+    @property
+    def supported_features(self) -> int:
+        """Return the list of supported features."""
+        return SUPPORT_FLAGS
 
     @property
-    def temperature_unit(self):
+    def temperature_unit(self) -> str:
         """Return the unit of measurement used by the platform."""
         return TEMP_CELSIUS
 
     @property
-    def current_operation(self):
+    def current_operation(self) -> str:
         """Return current operation i.e. comfort, home, away."""
-        return TOON_HA.get(self.thermos.get_data('state'))
+        return TOON_HA.get(self._state)
 
     @property
-    def operation_list(self):
+    def operation_list(self) -> List[str]:
         """Return a list of available operation modes."""
-        return self._operation_list
+        return list(HA_TOON.keys())
 
     @property
-    def current_temperature(self):
+    def current_temperature(self) -> float:
         """Return the current temperature."""
-        return self.thermos.get_data('temp')
+        return self._current_temperature
 
     @property
-    def target_temperature(self):
+    def target_temperature(self) -> float:
         """Return the temperature we try to reach."""
-        return self.thermos.get_data('setpoint')
+        return self._target_temperature
+
+    @property
+    def device_state_attributes(self) -> Dict[str, Any]:
+        """Return the current state of the burner."""
+        return {
+            'heating_type': self._heating_type,
```

Comment:
Is this like operation mode ?

Reply:
No, `heatingType` is the basic energy source used for heating the boiler. Will be `gas` for most people.